lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -1357,17 +1357,50 @@ namespace Js
 
         RecyclableObject* instance = RecyclableObject::FromVar(instanceVar);
         ScriptContext* scriptContext = instance->GetScriptContext();
+
+        if (!PHASE_OFF1(IsConcatSpreadableCachePhase))
+        {
+            BOOL retVal = FALSE;
+            Type *instanceType = instance->GetType();
+            IsConcatSpreadableCache *isConcatSpreadableCache = scriptContext->GetThreadContext()->GetIsConcatSpreadableCache();
+
+            if (isConcatSpreadableCache->TryGetIsConcatSpreadable(instanceType, &retVal))
+            {
+#ifdef ENABLE_DEBUG_CONFIG_OPTIONS
+                if (Js::Configuration::Global.flags.Trace.IsEnabled(Phase::IsConcatSpreadableCachePhase))
+                {
+                    Output::Print(_u("IsConcatSpreadableCache hit: %p\n"), instanceType);
+                }
+#endif
+                return retVal;
+            }
+
+            Var spreadable = JavascriptOperators::GetProperty(instance, PropertyIds::_symbolIsConcatSpreadable, scriptContext);
+
+            if (spreadable != scriptContext->GetLibrary()->GetUndefined())
```

Comment:
Yes, the spec (22.1.3.1.1) says:

3.If spreadable is not undefined, return ToBoolean(spreadable).
4.Return ? IsArray(O). 


---
In reply to: [68430889](https://github.com/Microsoft/ChakraCore/pull/1195#discussion_r68430889) [](ancestors = 68430889)